gravitee-apim-console-webui/src/portal/navigation-items/portal-navigation-items.component.ts

```diff
@@ -94,4 +120,61 @@ export class PortalNavigationItemsComponent implements OnInit {
     this.pageNotFound = true;
     this.snackBarService.error('The requested Navigation Item does not exist.');
   }
+
+  onAddPageSectionClick() {
+    this.manageSection('PAGE', 'create', 'TOP_NAVBAR');
+  }
+
+  private manageSection(type: PortalNavigationItemType, mode: SectionEditorDialogMode, area: PortalArea): void {
+    this.matDialog
+      .open<SectionEditorDialogComponent, SectionEditorDialogData>(SectionEditorDialogComponent, {
+        width: GIO_DIALOG_WIDTH.SMALL,
+        data: {
+          type,
+          mode,
+        },
+      })
+      .afterClosed()
+      .pipe(
+        filter((result) => !!result),
+        switchMap((result) => {
+          return this.createAndRefreshList({
+            title: result.title,
+            type,
+            area,
+            url: type === 'LINK' && result.settings ? result.settings.url : undefined,
+          });
+        }),
```

Comment:
What do you think about separating the create and refresh into 2 pipeable actions?

```suggestion
        switchMap((result) => {
          return this.create({
            title: result.title,
            type,
            area,
            url: type === 'LINK' && result.settings ? result.settings.url : undefined,
          });
        }),
        switchMap((item) => this.refreshList().pipe(map(() => item))),

Reply:
Done!